optuna/samplers/_ga/_base.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import abc
+from typing import Any
+
+import optuna
+from optuna.samplers._base import BaseSampler
+from optuna.trial._frozen import FrozenTrial
+from optuna.trial._state import TrialState
+
+
+# TODO(gen740): Add the experimental decorator?
+class BaseGASampler(BaseSampler, abc.ABC):
+    """Base class for Genetic Algorithm (GA) samplers.
+
+    Genetic Algorithm samplers generate new trials by mimicking natural selection, using
+    generations and populations to iteratively improve solutions. This base class defines the
+    interface for GA samplers in Optuna and provides utility methods for managing generations and
+    populations.
+
+    The selection process is handled by :meth:`~BaseGASampler.select_parent`, which must be
+    implemented by subclasses to define the parent selection strategy.
+
+    Generation and population management is facilitated by methods like
+    :meth:`~BaseGASampler.get_generation` and :meth:`~BaseGASampler.get_population`, ensuring
+    consistent tracking and selection.
+
+    Note:
+        This class should be extended by subclasses that define specific GA sampling strategies,
+        including parent selection and crossover operations.
+    """
+
+    _GENERATION_KEY = "BaseGASampler:generation"
+    _PARENT_CACHE_KEY_PREFIX = "BaseGASampler:parent:"
+
+    def __init_subclass__(cls, **kwargs: Any):
+        super().__init_subclass__(**kwargs)
+        cls._GENERATION_KEY = f"{cls.__name__}:generation"
+        cls._PARENT_CACHE_KEY_PREFIX = f"{cls.__name__}:parent:"
+
+    @classmethod
+    def _get_generation_key(cls) -> str:
+        return cls._GENERATION_KEY
+
+    @classmethod
+    def _get_parent_cache_key_prefix(cls) -> str:
+        return cls._PARENT_CACHE_KEY_PREFIX
+
+    def __init__(self, population_size: int):
```

Comment:
How about allowing `population_size` to be `None` in the constructor?
`population_size` is often automatically determined based on the dimensionality of the search space.
For instance, `CmaEsSampler` adopts such a mechanism as follows.
https://github.com/optuna/optuna/blob/46d7dab172ecb62ea3a0bad87e771ab25905f0ff/optuna/samplers/_cmaes.py#L388-L389